jib-plugins-common/src/main/java/com/google/cloud/tools/jib/plugins/common/PluginConfigurationProcessor.java

```diff
@@ -204,8 +230,105 @@ public static PluginConfigurationProcessor processCommonConfiguration(
       jibContainerBuilder.setCreationTime(Instant.now());
     }
 
+    PluginConfigurationProcessor.configureContainerizer(
+        containerizer, rawConfiguration, projectProperties);
+
     return new PluginConfigurationProcessor(
-        jibContainerBuilder, baseImageReference, optionalFromCredential.isPresent());
+        jibContainerBuilder,
+        containerizer,
+        baseImageReference,
+        targetImageReference,
+        isBaseImageCredentialPresent,
+        isTargetImageCredentialPresent);
+  }
+
+  /**
+   * Gets the value of the {@code appRoot} parameter. If the parameter is empty, returns {@link
+   * JavaLayerConfigurations#DEFAULT_WEB_APP_ROOT} for WAR projects or {@link
+   * JavaLayerConfigurations#DEFAULT_APP_ROOT} for other projects.
+   *
+   * @param rawConfiguration raw configuration data
+   * @param projectProperties used for providing additional information
+   * @return the app root value
+   * @throws AppRootInvalidException if {@code appRoot} value is not an absolute Unix path
+   */
+  @VisibleForTesting
+  static AbsoluteUnixPath getAppRootChecked(
+      RawConfiguration rawConfiguration, ProjectProperties projectProperties)
+      throws AppRootInvalidException {
+    String appRoot = rawConfiguration.getAppRoot();
+    if (appRoot.isEmpty()) {
+      appRoot =
+          projectProperties.isWarProject()
+              ? JavaLayerConfigurations.DEFAULT_WEB_APP_ROOT
+              : JavaLayerConfigurations.DEFAULT_APP_ROOT;
+    }
+    try {
+      return AbsoluteUnixPath.get(appRoot);
+    } catch (IllegalArgumentException ex) {
+      throw new AppRootInvalidException(appRoot, appRoot, ex);
+    }
+  }
+
+  @FunctionalInterface
+  private static interface InferredAuthProvider {
+
+    Optional<AuthProperty> getInferredAuth(String registry) throws InferredAuthRetrievalException;
+  };
+
+  private static boolean configureCredentialRetrievers(
```

Comment:
This would be good to refactor later so that it doesn't have such a large parameter list.

Reply:
Agreed, I'll add a TODO. The long list of parameters is a consequence of just trying to technically share some section of code. At least we can remove `knownAuthSource` once the NPE problem is fixed.

Reply:
Ah yea, I've been meaning to add the style for that so we can reference it.